### ISSUE TITLE
Add an effect_on_condition to add or extend sidebar widgets

### ISSUE THREAD

**Issue by wapcaplet** [CleverRaven/Cataclysm-DDA#54655]
### Is your feature request related to a problem? Please describe.

Some widgets are only relevant under certain conditions. For example, the "Bionic power" widget is not worth having around unless you have bionics, and the "Radiation badge" widget is pretty useless until you're wearing a radiation badge.

The player can enable these widgets themselves at the appropriate time, assuming they're aware of them. But we have a quite flexible [effect on condition](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/EFFECT_ON_CONDITION.md) system that may be a good fit for these use cases.

### Solution you would like.

An "effect" capable of altering or extending the widgets generating the "custom" sidebar. I've not worked with effects on conditions yet myself, but these use cases (bionics and radiation badge) could potentially be supported with the conditions we already have defined.

I suggest an effect name like `modify_widgets`, with its parameter being a JSON list of widget objects, and/or extend and delete operations on existing widgets defined in `data/json/ui`.

Rough example for the radiation badge (the `RAD_DETECT` flag having been added to this item in #53915):

```json
{
  "type": "effect_on_condition",
  "id": "show_worn_radiation_badge",
  "condition": { "u_has_worn_with_flag": "RAD_DETECT" },
  "effect": {
    "modify_widgets": [
      { "id": "rad_badge_desc", "delete": { "flags": [ "W_DISABLED" ] } }
    ]
  }
}
```

For bionics, perhaps something like:

```json
{
  "type": "effect_on_condition",
  "id": "show_bionic_power",
  "condition": { "u_has_bionics": "ANY" },
  "effect": {
    "modify_widgets": [
      { "id": "bionic_desc", "delete": { "flags": [ "W_DISABLED" ] } }
    ]
  }
}
```

Note, presently neither the "power_desc" nor "rad_badge_desc" widgets have the `W_DISABLED` flag by default, but with these effects in place, they ought to be disabled until needed. A similar effect could potentially be used to hide the widgets when they're just taking up space after the badge or bionic is lost.

These are simplistic examples of deleting a single flag from an existing widget, but the existing architecture in `src/dialogue.h` appears to support generic JSON as a parameter to the effect, so there should be rich opportunity for customizing how the sidebar appears, just using EOCs.

Other effects that might be applicable:

- When acquiring the `NOPAIN` mutation, alter the health bar widgets to show "Good" or "Bad" (or some other crude short-bar version)
- When starting or stopping driving a vehicle, show or hide the "Vehicle" section
- Option to hide body part widgets by default, and show them only when injured. Also, tentacles. (@Venera3)

### Describe alternatives you have considered.

I could dive in and try to learn all about EOCs myself, but everything I know about them, I learned from writing this issue. So I would much rather invite participation from developers who know the system a little better (or who are more eager to learn how to use it than I am).

**Comment by dseguin** [comment]
I like this idea!

Just a point of clarification:
`W_DISABLED` only makes the widget disabled for the first time that widget is loaded. Once the player loads the game, the widget config is saved to and loaded from panel_options.json. If you want to enable/disable widgets on the fly via EOCs, you need the EOC effect to toggle the `window_panel::toggle` var.

Also, if anyone feels like giving this a shot, feel free to pick up where I left off #54468. If no one bites, I can give it a second try.

**Comment by wapcaplet** [comment]
Oh I had forgotten you used EOCs for that too. Cool, I knew we’d be coming back to it sooner or later.

Thanks for clarifying the disabled flag, I wasn’t sure if it would work that way.

**Comment by Ramza13** [comment]
I'm thinking  a bunch about the best way to do this. Its not hard conceptually but the way EOC's are setup is a poor match right now.  

It would be easy to set it up to have a check run everyturn and set the widget value when changed.  But especially if this scales we don't want to have tons of checks running everyturn, what we want is to have checks run whenever traits/bionics or items worn change.  Right now we have a partial version of this in regards to traits/bionics with deactivate conditions.

I might end up adding a new type of EOC, or something, I'm still thinking.

**Comment by Inglonias** [comment]
This would also be nice because the classic JSON sidebar differs from the old C++ version in that the vehicle widgests do not, and cannot, replace widgets in the health section like they used to.

**Comment by mlange-42** [comment]
This behaviour could now be achieved using swappable content in layout widget, as implemented in #59442. 

This would, however, shift responsibility to sidebar designers and away from the EoC design. Not sure if this would be a good thing.
